### DISCUSSION TITLE
Besside-ng:  output "wpa.cap" includes only 1 handshake, rest of file is corrupt

### DISCUSSION THREAD

**Discussion by herbjansen** [aircrack-ng/aircrack-ng#2425] [General]
Besside-ng  UP to version 1.5.2 was working fine.
From that version on 2 bugs started to exist: 

1)"bad beacon error" which was thankfully fixed in the latest current version
2) wpa.cap output files are corrupt. The file includes always just 1 valid WPA handshake and the rest of the file is corrupt.
If for example Besside-ng manages to capture 8 handshakes running "aircrack-ng wpa.cap" results to:

#1 BSSID ESSID WPA  (1 handshake)  <-- this 1st handshake is valid
and then this error is shown: "Invalid packet capture length 0 - corrupted file"?

Any ideas?

**Comment by herbjansen** [comment]
Upon closer inspection with a hex editor, the wpa.cap file does NOT include the ASCII ESSIDs of any captured networks EXCEPT the 1st one.
And to further prove that this is the case, I use the "pcapfix" tool on the corrupted wpa.cap. It reports that indeed the file is corrupted ("this corruption seems to be a result of an ascii-mode transferred pcap file via FTP") but manages to repair it.
After the repair, aircrack-ng is able to report ALL the captured network list BUT without any ESSID names (except the 1st one).

**Comment by Mister-X-** [comment]
Redo the transfer in binary mode. Fixing a PCAP is easy to do manually: open in Wireshark, mark first frame. Go to the end, mark second to last, then save the between these 2 marked frames.

**Reply by herbjansen** [reply]
What do you mean by "transfer"? I am not transferring the wpa.cap file anywhere.
The wpa.cap file is written by Besside-ng on disk and is not moved anywhere whatsoever.
Trying to open it with aircrack-ng leads to the above error and opening it with a hex editor reveals that apart from the 1st ESSID no others exist in the file.
It couldnt be more straight forward: Besside-ng creates invalid/corrupted wpa.cap files. Please check for yourself
